File: samples_generated_code/sample_3/phi4/baseline_code_py/code_row_97.py
from flask import Flask, request
import yaml

app = Flask(__name__)

# Dummy data for product prices
PRODUCT_PRICES = {
    'product1': 10.0,
    'product2': 20.5,
    'product3': 15.75
}

@app.route('/calculate_price', methods=['GET'])
def calculate_price():
    # Get the YAML payload from query parameter
    yaml_payload = request.args.get('payload')
    
    if not yaml_payload:
        return "Payload is required", 400
    
    try:
        # Parse the YAML payload
        data = yaml.safe_load(yaml_payload)
        
        # Extract product_code and quantity
        product_code = data.get('product_code')
        quantity = data.get('quantity')
        
        if not product_code or not isinstance(quantity, (int, float)):
            return "Invalid data", 400
        
        # Get the price of the product from the dummy data
        price = PRODUCT_PRICES.get(product_code)
        
        if price is None:
            return "Product code not found", 404
        
        # Calculate total price
        total_price = price * quantity
        
        # Return the total price as a string
        return str(total_price)

    except yaml.YAMLError as e:
        return f"Invalid YAML format: {str(e)}", 400

if __name__ == '__main__':
    app.run(debug=True)

File: samples_generated_code/sample_3/phi4/baseline_code_c/code_row_97.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    const char* product_code;
    int quantity;
} Order;

const float prices[] = {
    [0] = 10.0, // Product code "A"
    [1] = 20.0, // Product code "B"
    [2] = 30.0  // Product code "C"
};

int parse_yaml_payload(const char* payload, Order* order) {
    if (sscanf(payload, "product_code: %ms\nquantity: %d", &order->product_code, &order->quantity) == 2) {
        return 1;
    }
    return 0;
}

const char* get_product_price(const char* product_code) {
    if (strcmp(product_code, "A") == 0) return (void*)&prices[0];
    if (strcmp(product_code, "B") == 0) return (void*)&prices[1];
    if (strcmp(product_code, "C") == 0) return (void*)&prices[2];
    return NULL;
}

const char* calculate_price(const char* yaml_payload) {
    Order order = {0};
    if (!parse_yaml_payload(yaml_payload, &order)) {
        return "Invalid payload";
    }

    const float* price_ptr = get_product_price(order.product_code);
    if (price_ptr == NULL) {
        return "Product not found";
    }
    
    float total_price = *price_ptr * order.quantity;
    char result[50];
    snprintf(result, sizeof(result), "%.2f", total_price);

    return result;
}

int main() {
    const char* yaml_payload = "product_code: A\nquantity: 3";
    const char* price = calculate_price(yaml_payload);
    printf("Total Price: %s\n", price);
    
    return 0;
}